src/flow_graph.py

```diff
@@ -1096,15 +1274,163 @@ def reset_block_info(self) -> None:
+                    # If the instruction is a function call and we don't have a source
+                    # for the argument, we can prune the argument from the input list.
+                    # Otherwise, this is likely undefined behavior in the original asm
+                    if ir.function_target is not None and inp in arch.argument_regs:
+                        ir.inputs.remove(inp)
+                    else:
+                        missing_regs.append((inp, ref))
+                inputs[inp] = sources
+
+            # Remove any clobbered accesses, or any MemoryAccesses that depend on
+            # clobbered Registers. Ex: If `$v0` is clobbered, remove `$v0` and `4($v0)`
+            for clob in ir.clobbers + ir.outputs:
+                for reg in reg_srcs:
+                    if access_depends_on(reg, clob) or access_may_overlap(reg, clob):
+                        reg_srcs.remove(reg)
+                        break
+
+            # Mark outputs as coming from this instruction
+            for out in ir.outputs:
+                reg_srcs[out] = RefSet([ref])
+
+        # Translate everything dominated by this node, now that we know our own
+        # final register flow_graph. This will eventually reach every node.
+        for child in node.immediately_dominates:
+            child_reg_srcs = reg_srcs.copy()
+            child_phis = AccessRefs()
+            flow_graph.node_phis[child] = child_phis
+
+            phi_regs = regs_clobbered_until_dominator(child)
+            for reg in phi_regs:
+                phi_reg_srcs = phi_reg_sources(child, reg, reg_srcs.get(reg))
+                # If phi_reg_srcs is invalid, then the reg is inconsistently set, so it should
+                # not be used by the child node.
+                # Otherwise, it *is* set in every control flow path to the child node, so
+                # it can be used (but it will have a phi value)
+                child_reg_srcs[reg] = phi_reg_srcs
+                child_phis[reg] = phi_reg_srcs
+
+            process_node(child, child_reg_srcs)
+
+    # Recursively traverse every node, starting with the entry node
+    # This populates in node_phis and instr_inputs
+    entry_node = flow_graph.entry_node()
+    entry_reg_srcs = AccessRefs.initial_function_regs(function.arguments, arch)
+    flow_graph.node_phis[entry_node] = AccessRefs()
+    process_node(entry_node, entry_reg_srcs)
+
+    # Populate instr_references for each instruction
+    for ref, inputs in flow_graph.instr_inputs.items():
+        for reg, deps in inputs.items():
+            for dep in deps:
+                if isinstance(dep, InstrRef):
+                    flow_graph.instr_references[dep].add(reg, ref)
+
+    if missing_regs:
+        print("/*")
+        print(f"Warning: in {function.name}, regs were read before being written to:")
+        for reg, ref in missing_regs:
+            print(f"   {reg} at {ref}: {ref.instruction()}")
+        print(f"*/")
```

Comment:
I loosely feel like since instruction inputs are kinda advisory anyway, given function calls, we can leave handling these kinds of errors to the translation phase. But it's not a strong opinion, warning here is fine too.